node-bb/README.md

```diff
@@ -0,0 +1,131 @@
+# TripleA Forums
+
+The software that powers the TripleA Forums is [NodeBB](https://github.com/NodeBB/NodeBB). 
+This document is instructions on how to build and run the Docker container of the forums using an 
+[official Docker container of NodeBB](https://hub.docker.com/r/nodebb/docker) as the base image.
+
+### Building
+
+1. Make a directory on your machine for where the config file and upload folder will live. ```mkdir -p /opt/triple-forums```
+1. Make a directory for uploaded files. Inside the upload directory, create folders category, emoji, files, profile, 
+sounds, and system.
+    ```
+    mkdir -p /opt/triple-forums/uploads/category
+    mkdir -p /opt/triple-forums/uploads/emoji
+    mkdir -p /opt/triple-forums/uploads/files
+    mkdir -p /opt/triple-forums/uploads/profile
+    mkdir -p /opt/triple-forums/uploads/sounds
+    mkdir -p /opt/triple-forums/uploads/system
+    ```
+1. Build the Docker image using the command ```docker build -t triplea-forums:latest .```
+
+### Running
+
+**First time run**
+
+You will need a running database. The database can be either Redis, Mongo, or Postgres. Make sure you have one of these
+installed on your machine or a Docker container running. 
+
+1. Run the triplea-forums:latest image using the command 
+```docker run -d --rm --name triplea-forums -v /opt/triple-forums/uploads:/usr/src/app/public/uploads -p 8080:4567 triplea-forums:latest```
+1. Navigate to http://0.0.0.0:8080/. This will be the install and configuration screen for NodeBB.
+1. Fill out the details on the page.
+1. Once the configuration is complete, you will need to save the config.json file off the container to the folder you 
```

Comment:
Can we mount the folder where the 'config.json' would land so that it is then stored on the file system? Is the config.json unique to the host machine that runs it? If we can't do a folder mount, I wonder if we can do a pre-fab config.json that is already ready to go.

Reply:
Unfortunately you can't mount the folder since it is the root of the application install. That would require you to duplicate the entire install on your host in that folder in order for the container to run, which defeats the need for Docker. 

The config.json is not unique. You can use it across multiple hosts and docker instances. The real issue is the database that NodeBB uses from the config.json file. NodeBB uses that setup process to initialize the database with tables and the first admin user. The good thing is that you only have to run that setup once. After the database is setup, you are free to do whatever.

Reply:
Cool, we'll have ansible deploy the container, we can also use ansible to run any setup commands as well, same for local (future steps). Ideally we can get this all to be as 'one-touch' as possible.